### DISCUSSION TITLE
Alt+Tab on remote: is there anything else beyond grabbing the keyboard?

### DISCUSSION THREAD

**Discussion by charlescanato** [Xpra-org/xpra#3748] [Q&A] [answered]
This is my first day with xpra - and what a piece of software! However, Alt+Tab, by what I have read around, should be working. After I grab the keyboard (Ctrl+Menu), indeed Alt+Tab is not sent to client (Win7) - and it shouldn't, but remote host behaves as if has only received a Tab.

I didn't want to create this as a bug, as I'm pretty sure I'm one of the few (if any) who can't use Alt+tab to go back and forth from my other applications on the client. I just didn't notice yet what I might be doing wrong.

**To Reproduce**
1. `xpra shadow :0 --bind-tcp=0.0.0.0:18500 --daemon=no --opengl=no --speaker=off --microphone=off` (first times I just used the simple shadow command directly from the client)
2. `xpra_cmd.exe --env=XPRA_EMULATE_ALTGR=0 attach tcp://192.168.0.23:18500 --opengl=no
 --speaker=off --microphone=off` (I got the env from some other issue, while trying anything - I also turned off sound and OpenGL based on some tests I've saw in other issues)
3. Ubuntu desktop window is opened totally ok - I have been using Xpra's toolbox "Keyboard State Tool" for my tests
4. Grab the keyboard, if not already using Ctrl+Menu
5. Press and hold Alt-L
6. ![](https://i.imgur.com/nzw2LML.png)
7. Press and release Tab
8. ![](https://i.imgur.com/0nUni4Y.png)
9. As can be seen in the screenshot, before Tab goes down, Alt is released, and there's also a ghost Control_L added - which, of course, was never pressed

**System Information (please complete the following information):**
 - Server OS: Ubuntu 22.04 LTS with Kubuntu
 - Client OS: Windows 7
 - Xpra Server Version: v5.0-r32495 beta
 - Xpra Client Version: v5.0-r32489 (initially used v4.4.3-r0, same behavior)

I have AutoHotKey running a few keys - closing it and restarting everything also does not change the scenario.

Is there something really basic I'm missing?

**Comment by totaam** [comment, marked as answer]
That should work.
> indeed Alt+Tab is not sent to client (Win7) - and it shouldn't

The fact that `Alt+Tab` is intercepted is a good sign.
Don't use `XPRA_EMULATE_ALTGR=0`, it will cause extra keys to be pressed / removed.
The spurious `Control_L` key events are a "feature" of key events on MS Windows.
Can you post the server log when running with `-d keyboard`?
(just the bit where you press `Alt+Tab` as it will be huge otherwise)

---

@charlescanato Edit: oh wait, you're using `shadow` mode on the server... I have never tested that.
You may be able to get it to work with `--keyboard-layout=auto` on the server, as per
https://github.com/Xpra-org/xpra/issues/2630#issuecomment-1193995389
